crates/outbound-redis/src/lib.rs

```diff
@@ -3,13 +3,33 @@ mod host_component;
 use std::collections::{hash_map::Entry, HashMap};
 
 use anyhow::Result;
-use redis::{aio::Connection, AsyncCommands};
+use redis::{aio::Connection, AsyncCommands, FromRedisValue, RedisResult, Value};
 use wit_bindgen_wasmtime::async_trait;
 
 pub use host_component::OutboundRedisComponent;
 
 wit_bindgen_wasmtime::export!({paths: ["../../wit/ephemeral/outbound-redis.wit"], async: *});
-use outbound_redis::Error;
+use outbound_redis::{Error, ValueParam, ValueResult};
+
+struct Values(Vec<ValueResult>);
+
+impl FromRedisValue for Values {
+    fn from_redis_value(value: &Value) -> RedisResult<Self> {
+        fn append(values: &mut Vec<ValueResult>, value: &Value) {
+            match value {
+                Value::Nil | Value::Okay => (),
+                Value::Int(v) => values.push(ValueResult::Int(*v)),
+                Value::Data(bytes) => values.push(ValueResult::Data(bytes.to_owned())),
+                Value::Bulk(bulk) => bulk.iter().for_each(|value| append(values, value)),
+                Value::Status(message) => values.push(ValueResult::String(message.to_owned())),
```

Comment:
If I understand this correctly, it will only return a `ValueResult::String` when a status is returned? I used list operations that return strings (`lpush` and `lrange`), and was surprised that the results were always `ValueResult::Data` that needed to be decoded. The only ask here is that we document what operation is expected to return a string.

Reply:
Yeah, I initially assumed more things would return strings and integers, but apparently Redis really likes returning byte arrays, leaving it up to the client to parse the expected type.  I agree that renaming it to `status` makes sense.